### DISCUSSION TITLE
How to declare and use custom local types?

### DISCUSSION THREAD

**Discussion by siarheipashkevich** [larastan/larastan#1773] [Q&A] [answered]
I'm going to strict my types but I don't understand why it is incorrect?

![image](https://github.com/phpstan/phpstan/assets/7753600/9df45345-31b3-4fe3-bbfa-b73494176f25)

![image](https://github.com/phpstan/phpstan/assets/7753600/e88ddb49-106c-47b1-af80-648a7c831c70)

![image](https://github.com/phpstan/phpstan/assets/7753600/c7319f1b-ac6a-491d-b389-08f7145fe65f)

Thanks

**Comment by szepeviktor** [comment]
That is good question.
Are interested in IDE usage, Larastan usage or [PHPStan usage](https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases)?

**Comment by mad-briller** [comment, marked as answer]
phpstorm doesn't support `@phpstan-type` yet (?) so you're out of luck
